memcache/memcache.go

```diff
@@ -233,12 +243,13 @@ func (cn *conn) extendDeadline() {
 // is nil (not an error) or is only a protocol level error (e.g. a
 // cache miss).  The purpose is to not recycle TCP connections that
 // are bad.
-func (cn *conn) condRelease(err *error) {
+func (cn *conn) condRelease(err *error, lock chan struct{}) {
 	if *err == nil || resumableError(*err) {
 		cn.release()
 	} else {
 		cn.nc.Close()
 	}
+	<-lock
```

Comment:
should this be in a `defer`?

Reply:
yeah, I guess.  easier than freaking out about panics on release or Close()